source/tyr/tyr/resources.py

```diff
@@ -701,3 +726,69 @@ def delete(self, name=None, traveler_type=None):
         except Exception:
             logging.exception("fail")
             raise
+
+
+class Plan(flask_restful.Resource):
+    def __init__(self):
+        pass
+
+    def get(self, plan_id=None):
+        if plan_id:
+            plan = models.Plan.query.get_or_404(plan_id)
+            return marshal(plan, plan_fields)
+        else:
+            plans = models.Plan.query.all()
+            return marshal(plans, plan_fields)
+
+    def post(self):
+        parser = reqparse.RequestParser()
+        parser.add_argument('name', type=unicode, required=True,
+                            case_sensitive=False, help='name is required',
+                            location=('json', 'values'))
+        parser.add_argument('max_request_count', type=int, required=False,
+                            help='max request count for this plan', location=('json', 'values'))
+        parser.add_argument('max_object_count', type=int, required=False,
+                            help='max object count for this plan', location=('json', 'values'))
+        args = parser.parse_args()
+
+        try:
+            plan = models.Plan(name=args['name'], max_request_count=args['max_request_count'],
+                               max_object_count=args['max_object_count'])
+            db.session.add(plan)
+            db.session.commit()
+            return marshal(plan, plan_fields)
+        except Exception:
+            logging.exception("fail")
+            raise
+
+    def put(self, plan_id=None):
+        plan = models.Plan.query.get_or_404(plan_id)
+        parser = reqparse.RequestParser()
```

Comment:
It's seem to be the same parser used in post, you can create it in the ```_init__``` as a instance's attribute